system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp

```diff
@@ -0,0 +1,587 @@
+    auto result = client_mrm_pull_over_->async_send_request(request).get();
+    if (result->response.success == true) {
+      RCLCPP_WARN(this->get_logger(), "Pull over is canceled");
+    } else {
+      RCLCPP_ERROR(this->get_logger(), "Pull over failed to cancel");
+    }
+    return;
+  }
+  if (mrm_behavior == MrmState::COMFORTABLE_STOP) {
+    auto result = client_mrm_comfortable_stop_->async_send_request(request).get();
+    if (result->response.success == true) {
+      RCLCPP_WARN(this->get_logger(), "Comfortable stop is canceled");
+    } else {
+      RCLCPP_ERROR(this->get_logger(), "Comfortable stop failed to cancel");
+    }
+    return;
+  }
+  if (mrm_behavior == MrmState::EMERGENCY_STOP) {
+    auto result = client_mrm_emergency_stop_->async_send_request(request).get();
+    if (result->response.success == true) {
+      RCLCPP_WARN(this->get_logger(), "Emergency stop is canceled");
+    } else {
+      RCLCPP_ERROR(this->get_logger(), "Emergency stop failed to cancel");
+    }
+    return;
+  }
+  RCLCPP_WARN(this->get_logger(), "invalid MRM behavior: %d", mrm_behavior);
+}
+
+bool MrmHandler::isDataReady()
+{
+  if (!operation_mode_availability_) {
+    RCLCPP_INFO_THROTTLE(
+      this->get_logger(), *this->get_clock(), std::chrono::milliseconds(5000).count(),
+      "waiting for operation_mode_availability msg...");
+    return false;
+  }
+
+  if (
+    param_.use_pull_over &&
+    mrm_pull_over_status_->state == tier4_system_msgs::msg::MrmBehaviorStatus::NOT_AVAILABLE) {
+    RCLCPP_INFO_THROTTLE(
+      this->get_logger(), *this->get_clock(), std::chrono::milliseconds(5000).count(),
+      "waiting for mrm pull over to become available...");
+    return false;
+  }
+
+  if (
+    param_.use_comfortable_stop && mrm_comfortable_stop_status_->state ==
+                                     tier4_system_msgs::msg::MrmBehaviorStatus::NOT_AVAILABLE) {
+    RCLCPP_INFO_THROTTLE(
+      this->get_logger(), *this->get_clock(), std::chrono::milliseconds(5000).count(),
+      "waiting for mrm comfortable stop to become available...");
+    return false;
+  }
+
+  if (
+    mrm_emergency_stop_status_->state == tier4_system_msgs::msg::MrmBehaviorStatus::NOT_AVAILABLE) {
+    RCLCPP_INFO_THROTTLE(
+      this->get_logger(), *this->get_clock(), std::chrono::milliseconds(5000).count(),
+      "waiting for mrm emergency stop to become available...");
+    return false;
+  }
+
+  return true;
+}
+
+void MrmHandler::onTimer()
+{
+  if (!isDataReady()) {
+    return;
+  }
+  const bool is_operation_mode_availability_timeout =
+    (this->now() - stamp_operation_mode_availability_).seconds() >
+    param_.timeout_operation_mode_availability;
+  if (is_operation_mode_availability_timeout) {
+    RCLCPP_WARN_THROTTLE(
+      this->get_logger(), *this->get_clock(), std::chrono::milliseconds(1000).count(),
+      "heartbeat operation_mode_availability is timeout");
+    mrm_state_.state = autoware_adapi_v1_msgs::msg::MrmState::MRM_OPERATING;
```

Comment:
Are you going to apply this change later?
https://github.com/autowarefoundation/autoware.universe/pull/6399

Reply:
Yes. This PR just replaces the topic.